### PR TITLE
Refactored tests for decorators

### PR DESCRIPTION
**Pull request by pdthummar**

### CODE REVIEW

File: tests/decorators/test_decorators.py [hunk 1]
```diff
@@ -201,6 +201,48 @@ def dummy():
...
+        app = self.func_app
+
+        @app.on_queue_change(arg_name="req", queue_name="dummy_queue",
+                             connection="dummy_conn")
+        def dummy():
+            pass
+
+        func = self._get_func(app)
+
+        trigger = func.get_bindings()[0]
```
Comment (YunchuWang) [inline tests/decorators/test_decorators.py:214]: minor: add binding list length check, for all of tests

File: tests/decorators/test_decorators.py [hunk 2]
```diff
@@ -201,6 +201,48 @@ def dummy():
...
+        @app.on_queue_change(arg_name="req", queue_name="dummy_queue",
+                             connection="dummy_conn")
+        @app.write_queue(arg_name="out", queue_name="dummy_out_queue",
+                         connection="dummy_out_conn")
+        def dummy():
+            pass
+
+        func = self._get_func(app)
+
+        trigger = func.get_bindings()[0]
```
Comment (YunchuWang) [inline tests/decorators/test_decorators.py:236]: this is testing the output, please rename to output.

File: tests/decorators/test_decorators.py [hunk 3]
```diff
@@ -267,6 +309,51 @@ def dummy():
...
+                                         queue_name="dummy_queue")
+        @app.write_service_bus_queue(arg_name='res',
+                                     connection='dummy_out_conn',
+                                     queue_name='dummy_out_queue')
+        def dummy():
+            pass
+
+        func = self._get_func(app)
+
+        trigger = func.get_bindings()[0]
```
Comment (YunchuWang) [inline tests/decorators/test_decorators.py:347]: ditto, and change to output for any remaining tests

File: tests/decorators/test_decorators.py [hunk 4]
```diff
@@ -348,6 +435,55 @@ def dummy():
                                  ]
                                  })
 
+    def test_service_bus_trigger(self):
```
Comment (gavin-aguiar) [inline tests/decorators/test_decorators.py:438]: test_service_bus_topic_trigger

File: tests/decorators/test_decorators.py [hunk 5]
```diff
@@ -348,6 +435,55 @@ def dummy():
...
+        self.assertEqual(trigger.get_dict_repr(), {
+            "type": SERVICE_BUS_TRIGGER,
+            "direction": BindingDirection.IN,
+            "name": "req",
+            "connection": "dummy_conn",
+            "topicName": "dummy_topic",
+            "subscriptionName": "dummy_sub"
+        })
+
+    def test_service_bus_output_binding(self):
```
Comment (gavin-aguiar) [inline tests/decorators/test_decorators.py:461]: topic_output_binding

File: tests/decorators/test_decorators.py [hunk 6]
```diff
@@ -201,6 +201,48 @@ def dummy():
...
+
+        self.assertEqual(trigger.get_dict_repr(), {
+            "direction": BindingDirection.IN,
+            "type": QUEUE_TRIGGER,
+            "name": "req",
+            "queueName": "dummy_queue",
+            "connection": "dummy_conn"
+        })
+
+    def test_queue_trigger_output_binding(self):
```
Comment (gavin-aguiar) [inline tests/decorators/test_decorators.py:224]: Remove the trigger from the name as this test is only testing the output binding. Similar comment for the other tests too

File: tests/decorators/test_decorators.py (hunk 1 above)
Comment (pdthummar) [reply tests/decorators/test_decorators.py]: updated as per recommendation. Thanks

File: tests/decorators/test_decorators.py (hunk 2 above)
Comment (pdthummar) [reply tests/decorators/test_decorators.py]: I was matching the test_blob_trigger for this. I have renamed as per recommendation. Thanks

File: tests/decorators/test_decorators.py (hunk 3 above)
Comment (pdthummar) [reply tests/decorators/test_decorators.py]: I have renamed this for all tests as per recommendation. Thanks

File: tests/decorators/test_decorators.py (hunk 4 above)
Comment (pdthummar) [reply tests/decorators/test_decorators.py]: updated as per recommendation. Thanks

File: tests/decorators/test_decorators.py (hunk 5 above)
Comment (pdthummar) [reply tests/decorators/test_decorators.py]: updated as per recommendation. Thanks

File: tests/decorators/test_decorators.py (hunk 6 above)
Comment (pdthummar) [reply tests/decorators/test_decorators.py]: updated as per recommendation. Thanks

File: tests/decorators/test_decorators.py [hunk 7]
```diff
@@ -211,17 +211,18 @@ def dummy():
 
         func = self._get_func(app)
 
-        trigger = func.get_bindings()[0]
+        output = func.get_bindings()[0]
 
-        self.assertEqual(trigger.get_dict_repr(), {
+        self.assertEqual(len(func.get_bindings()), 1)
```
Comment (YunchuWang) [inline tests/decorators/test_decorators.py:216]: minor: put length check before output to not cause null pointer exception, else looks good!

Comment (pdthummar) [reply tests/decorators/test_decorators.py:214]: updated Thanks.

**Review by gavin-aguiar — APPROVED** [review]

File: tests/decorators/test_decorators.py [hunk 8]
```diff
@@ -201,6 +201,50 @@ def dummy():
...
 
+    def test_queue_trigger(self):
+        app = self.func_app
+
+        @app.on_queue_change(arg_name="req", queue_name="dummy_queue",
+                             connection="dummy_conn")
+        def dummy():
+            pass
+
+        func = self._get_func(app)
```
Comment (vrdmr) [inline tests/decorators/test_decorators.py:212]: Can you please change the name of this helper function - calling it `_get_user_function`?

**Review by vrdmr — CHANGES_REQUESTED** [review]

File: tests/decorators/test_decorators.py [hunk 9]
```diff
@@ -201,6 +201,50 @@ def dummy():
                                          "connection": "dummy_conn"
                                      }]})
 
+    def test_queue_trigger(self):
+        app = self.func_app
+
+        @app.on_queue_change(arg_name="req", queue_name="dummy_queue",
```
[the remark this answers is not in the document]
Comment (vrdmr) [reply tests/decorators/test_decorators.py:207]: Requesting changes for this. Will clear post #112 and you update the PR. Thanks.

File: tests/decorators/test_decorators.py (hunk 8 above)
Comment (pdthummar) [reply tests/decorators/test_decorators.py]: updated as per recommendation. Thanks.

**Comment by YunchuWang** [comment]
please resolve conflicts

**Review by YunchuWang — APPROVED** [review]

**Pull request merged**